### PR TITLE
Prevent errors when non-string CSP values are defined in the manifest

### PR DESCRIPTION
**Pull request by willdurand**
Fixes https://github.com/mozilla/addons-linter/issues/5194

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/ADDLINT-462)

### CODE REVIEW

File: tests/unit/parsers/test.manifestjson.js [hunk 1]
```diff
@@ -2248,6 +2248,24 @@ describe('ManifestJSONParser', () => {
         expect(warningsV3.length).toEqual(6);
       }
     );
+
+    // See: https://github.com/mozilla/addons-linter/issues/5194
+    it('should handle non-string values', () => {
+      const addonLinter = new Linter({ _: ['bar'] });
+      const json = validManifestJSON({ content_security_policy: [true] });
```
Comment (rpl) [inline tests/unit/parsers/test.manifestjson.js:2255]: @willdurand would you mind to also add a test manifest with a non string csp value when content_security_policy is in the object form? (e.g. even just making this same test executed for both this value and `{ content_security_policy: { extension_pages: true } }`).

File: src/parsers/manifestjson.js [hunk 2]
```diff
@@ -1149,6 +1149,10 @@ export default class ManifestJSONParser extends JSONParser {
   }
 
   validateCspPolicyString(policy, manifestPropName) {
+    if (typeof policy !== 'string') {
```
Comment (rpl) [inline src/parsers/manifestjson.js:1152]: @willdurand I wonder if we should also tweak the caller (`validateCspPolicy') to only iterate over `Object.keys(policy)` if policy is not null and not an array (given that for the other primitive types but array, Object.keys would already be an empty array and not doing anything in practice), the tweak in this PR should already prevent the kind of unexpected issues we may be hitting but in the end if the csp value is an array then it is already invalid on its own and validating each element of the array not strictly necessary anymore.

**Review by rpl — APPROVED** [review]
@willdurand the fix looks reasonable as is, the inline comments are related to a small addition to the test case and considering not validating the csp strings at all if the value of the content_security_policy field is an array (mainly to avoid to validate values that would be in an invalid form anyway).

File: src/parsers/manifestjson.js (hunk 2 above)
Comment (willdurand) [reply src/parsers/manifestjson.js:1153]: I added a test case to cover `null`.

**Pull request merged**